File: java-angular/book-social-network/book-network-api/src/main/java/koula/diallo/book/feedback/FeedbackRequest.java
package koula.diallo.book.feedback;

import jakarta.validation.constraints.*;

public record FeedbackRequest(
        @Positive(message = "200")
        @Min(value = 0, message = "201")
        @Max(value = 5, message = "202")
        Double note,
        @NotNull(message = "2003")
        @NotEmpty(message = "2003")
        @NotBlank(message = "2003")
        String comment,
        @NotNull(message = "204")
        Integer bookId
) {
}

File: mobile-apps/book_social_network/lib/models/feedback_request.dart
class FeedbackRequest {
  final double note;
  final String comment;
  final int bookId;

  FeedbackRequest(
      {required this.note, required this.comment, required this.bookId});

  Map<String, dynamic> toJson() {
    return {
      'note': note,
      'comment': comment,
      'bookId': bookId,
    };
  }
}
